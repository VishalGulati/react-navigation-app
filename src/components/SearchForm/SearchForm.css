.left-panel {
  text-align: left;
  margin-bottom: 10px;
}

.lp-btn + .lp-btn {
  margin-left: 10px;
}
