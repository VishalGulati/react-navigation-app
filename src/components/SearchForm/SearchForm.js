import React, { Component } from 'react';
import './SearchForm.css';
import PropTypes from 'prop-types';
import FormGroup from '../FormGroup/FormGroup';
import Message from '../Message/Message';

/**
 * The Left panel of the application that contains the search form.
 */
class SearchForm extends Component {
  /**
 * @name componentDidUpdate
 * @description React component lifecycle hook
 */
  componentDidUpdate() {
    if (this.props.resetPending) {
      this.myFormRef.reset();
      this.props.resetDone();
    }
  }

  /**
     * @name getBtnLabel
     * @description This method returns the label to be displayed on submit button, by default its Submit,
     * once form submitted, it will be Re-Submit and during API request it will be Loading...
     */
  getBtnLabel = () => {
    if (this.props.isLoading) {
      return 'Loading...';
    } else if (this.props.isDirty) {
      return 'Re-Submit';
    } else {
      return 'Submit';
    }
  };

  render() {
    return (
      <div className="col-xs-12 col-md-4 left-panel">
        <form
          onSubmit={this.props.handleSubmit}
          ref={el => (this.myFormRef = el)}
        >
          <FormGroup inputId="start" label="Starting location" />
          <FormGroup inputId="drop" label="Drop-off point" />
          <Message
            message={this.props.message}
            messageType={this.props.messageType}
          />
          <button
            type="button"
            className="btn btn-primary lp-btn"
            disabled={this.props.isLoading}
            onClick={this.props.handleSubmit}
          >
            {this.getBtnLabel()}
          </button>
          <button
            type="button"
            className="btn btn-secondary lp-btn"
            onClick={this.props.handleReset}
          >
            Reset
          </button>
        </form>
      </div>
    );
  }
}

SearchForm.propTypes = {
  message: PropTypes.string,
  messageType: PropTypes.string,
  isLoading: PropTypes.bool,
  isDirty: PropTypes.bool,
  mapLoaded: PropTypes.bool,
  handleSubmit: PropTypes.func,
  handleReset: PropTypes.func,
  resetDone: PropTypes.func,
  resetPending: PropTypes.bool
};

export default SearchForm;
