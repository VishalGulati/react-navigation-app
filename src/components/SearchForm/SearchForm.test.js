import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

const wrapper = shallow(<SearchForm />);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Component has container div with class left-panel', () => {
  expect(wrapper.find('div.left-panel').length).toEqual(1);
});

it('Component has a form element', () => {
  expect(wrapper.find('form').length).toEqual(1);
});

it('Component has 2 buttons', () => {
  expect(wrapper.find('button').length).toEqual(2);
});

it('snapshot testing', () => {
  expect(wrapper).toMatchSnapshot();
});
