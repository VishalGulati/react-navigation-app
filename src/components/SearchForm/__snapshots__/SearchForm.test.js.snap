// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <FormGroup
          inputId="start"
          label="Starting location"
        />
        <FormGroup
          inputId="drop"
          label="Drop-off point"
        />
        <Message />
        <button
          className="btn btn-primary lp-btn"
          disabled={false}
          type="button"
        />
        <button
          className="btn btn-secondary lp-btn"
          type="button"
        >
          Reset
        </button>
      </form>,
      "className": "col-xs-12 col-md-4 left-panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            inputId="start"
            label="Starting location"
          />,
          <FormGroup
            inputId="drop"
            label="Drop-off point"
          />,
          <Message />,
          <button
            className="btn btn-primary lp-btn"
            disabled={false}
            type="button"
          />,
          <button
            className="btn btn-secondary lp-btn"
            type="button"
          >
            Reset
          </button>,
        ],
        "onSubmit": undefined,
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputId": "start",
            "label": "Starting location",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputId": "drop",
            "label": "Drop-off point",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": undefined,
            "messageType": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "btn btn-primary lp-btn",
            "disabled": false,
            "onClick": undefined,
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset",
            "className": "btn btn-secondary lp-btn",
            "onClick": undefined,
            "type": "button",
          },
          "ref": null,
          "rendered": "Reset",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <FormGroup
            inputId="start"
            label="Starting location"
          />
          <FormGroup
            inputId="drop"
            label="Drop-off point"
          />
          <Message />
          <button
            className="btn btn-primary lp-btn"
            disabled={false}
            type="button"
          />
          <button
            className="btn btn-secondary lp-btn"
            type="button"
          >
            Reset
          </button>
        </form>,
        "className": "col-xs-12 col-md-4 left-panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              inputId="start"
              label="Starting location"
            />,
            <FormGroup
              inputId="drop"
              label="Drop-off point"
            />,
            <Message />,
            <button
              className="btn btn-primary lp-btn"
              disabled={false}
              type="button"
            />,
            <button
              className="btn btn-secondary lp-btn"
              type="button"
            >
              Reset
            </button>,
          ],
          "onSubmit": undefined,
        },
        "ref": [Function],
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "inputId": "start",
              "label": "Starting location",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "inputId": "drop",
              "label": "Drop-off point",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "message": undefined,
              "messageType": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "btn btn-primary lp-btn",
              "disabled": false,
              "onClick": undefined,
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset",
              "className": "btn btn-secondary lp-btn",
              "onClick": undefined,
              "type": "button",
            },
            "ref": null,
            "rendered": "Reset",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
